Lib/Search/SearchManager.php

```diff
@@ -278,12 +278,15 @@ public function remove($objects)
      *
      * @param mixed $object
      * @param bool $refresh
+     * @param bool $forceVersion
      *
      * @throws Exception\DoctrineSearchException
      */
-    public function flush($object = null, $refresh = false)
+    public function flush($object = null, $refresh = false, $forceVersion = false)
     {
+        $this->client->setForceVersion($forceVersion);
```

Comment:
I would prefer not adding a new argument here, but letting the caller to explicitly fetch the client and set the forceVersion and manually set it back.

Besides, need to reset the forceVersion once the searchManager is cleared. The cleared searchManager is supposed to be in its init condition.